### ISSUE TITLE
HeadRemove Bad Response & Include no longer supported in ListenHTTPS section

### ISSUE THREAD

**Issue by orbiiii** [graygnuorg/pound#5]
Hi

FreeBSD ports recently upgraded from pound 2.8 to 4.5 - and I ran across two issues with this.

1) The directive 'HeadRemove "X-Forwarded-Proto"' now results in a 'Bad Request' response when trying to access any websites.

2) 'Include' is no longer supported within ListenHTTPS.

For the latter, I was able to fix this with the following patch:
```
*** src/config.c.old    Sun Feb 12 19:56:35 2023
--- src/config.c        Sat Mar 18 20:02:12 2023
***************
*** 3387,3392 ****
--- 3387,3393 ----
  }

  static PARSER_TABLE https_parsetab[] = {
+   { "Include", parse_include },
    { "End", parse_end },
    { "Address", assign_address, NULL, offsetof (LISTENER, addr) },
    { "Port", assign_port, NULL, offsetof (LISTENER, addr) },
```

But a more generic case may be considered - to apply this to other sections within the configuration - but this was the only area I needed it fixed.

I did not do any further investigation on the HeadRemove issue.

**Comment by graygnuorg** [comment]
Hi,

Re 'HeadRemove' - I cannot reproducethis, neither on 4.5 nor on the current head. More detail will be appreciated.

Re 'Include' - agreed.

**Comment by orbiiii** [comment]
The following is the configuration that I'm using that seems to trigger it.  It's a fairly simple SSL reverse proxy that was previously working on Pound-2.8.  I've manually added the 'Include' support, and had to change the xHTTP from 4 to 3.

If I comment out the HeadRemove - things load fine - otherwise, I get a BadResponse from the client.  It seemed a bit odd to me, but I haven't looked much further into the cause - I just wanted my server back online.

`TimeOut         3600
LogLevel        5

IgnoreCase      1

ListenHTTPS
        Address <<IP>>
        Port    443

        Include "/usr/local/certs/pound.cfg"

        xHTTP   3

        RewriteLocation 0

        HeadRemove "X-Forwarded-Proto"
        AddHeader "X-Forwarded-Proto: https"

        Disable SSLv2
        Disable SSLv3
        Disable TLSv1
        Disable TLSv1_1

        SSLHonorCipherOrder 1

        # My compromise of ciphers, leaving CBC enabled
        # Note: Add ':!SHA1:!SHA256:!SHA384:!DSS:!aNULL' to disable
        Ciphers "ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-AES
128-GCM-SHA256:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-
GCM-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES256-SHA384:ECD
HE-RSA-AES256-SHA384"

        Service
                BackEnd
                        Address <<IP>>
                        Port 80
                End
        End
End
`

**Comment by graygnuorg** [comment]
Ah, yes, now I've got it.  There was a bug in the http_header_list_filter that was fixed in commit 3cd8178c85bcad723446f5ca0340f327dc86d139. The fix is included in version 4.6. Can you try that version?

**Comment by orbiiii** [comment]
I have manually updated the FreeBSD port to fetch 4.6 instead of 4.5 - and deployed that successfully.

I can confirm that the HeadRemove directive is working as expected on this version.

I will notify the FreeBSD port maintainer to update to 4.6.

**Comment by graygnuorg** [comment]
Thanks for testing this. I'll try to fix the 'include' issue asap and will post a notice here once it is done. 

> I will notify the FreeBSD port maintainer to update to 4.6, unless this is you

Nope, it's a friend of mine :) I believe he is monitoring the issues, but I'll notify him just in case.

**Comment by danfe** [comment]
> I will notify the FreeBSD port maintainer to update to 4.6

No need to. :-)  I'm working on the update now.

**Comment by graygnuorg** [comment]
The include issue has been fixed in commit 6c7258cb2e715994e01a1c4dd537a0f593cf7896.
